### ISSUE TITLE
route_settings unsupported in HTTP API Gateway

### ISSUE THREAD

**Issue by askaribragimov** [terraform-aws-modules/terraform-aws-apigateway-v2#42]
## Description
 
Current (1.1.0) version of the api-gateway module has this block:

```
  #  # bug - https://github.com/terraform-providers/terraform-provider-aws/issues/12893
  #  dynamic "route_settings" {
  #    for_each = var.create_routes_and_integrations ? var.integrations : {}
  #    content {
  #      route_key = route_settings.key
  #      data_trace_enabled = lookup(route_settings.value, "data_trace_enabled", null)
  #      detailed_metrics_enabled         = lookup(route_settings.value, "detailed_metrics_enabled", null)
  #      logging_level         = lookup(route_settings.value, "logging_level", null)  # Error: error updating API Gateway v2 stage ($default): BadRequestException: Execution logs are not supported on protocolType HTTP
  #      throttling_burst_limit         = lookup(route_settings.value, "throttling_burst_limit", null)
  #      throttling_rate_limit         = lookup(route_settings.value, "throttling_rate_limit", null)
  #    }
  #  }
```
TF documentation tells that stage's routes can have custom settings ( see "route_settings" in https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/apigatewayv2_stage)

As such, I can not do the following instruction, it has no effect:

```
module "api_gateway" {
  source  = "terraform-aws-modules/apigateway-v2/aws"
  version = "~> 1.1.0"
  protocol_type = "HTTP"
 # ...  options abriged...
  integrations = {
    "GET /auth/myinegration" = {
      #...abriged...
      # THESE INSTRUCTIONS ARE IN https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/apigatewayv2_stage but they are not supported by your module due to the commented bit quoted above.
      throttling_rate_limit=10
      throttling_burst_limit=10
    }
```

The issue https://github.com/terraform-providers/terraform-provider-aws/issues/12893 seem to be closed but it is hard to say if underlying issue is been fixed.

I am interested into setting `throttling_burst_limit` / `throttling_rate_limit` in particular.

**Comment by akubala** [comment]
I am also curious about the state of mentioned issue.
It would be great to use `data_trace_enabled` option for `default_route_settings` resource. :)

**Comment by antonbabenko** [comment]
This issue has been resolved in version 5.0.0 :tada:
